CMakeLists.txt

```diff
@@ -10,11 +12,41 @@ if(${CMAKE_PROJECT_NAME} STREQUAL ${PROJECT_NAME})
     set_property(GLOBAL PROPERTY PREDEFINED_TARGETS_FOLDER "CMake")
     set(EXECUTABLE_OUTPUT_PATH ${CMAKE_BINARY_DIR}/bin)
     set(LIBRARY_OUTPUT_PATH ${CMAKE_BINARY_DIR}/lib)
+else()
+    # Check required c++ standard of parent project
+    if(CMAKE_CXX_STANDARD)
+        set(PARENT_CXX_STANDARD ${CMAKE_CXX_STANDARD})
+        message(STATUS "CXX standard of parent project: ${PARENT_CXX_STANDARD}")
+    endif()
 endif()
 
 # Avoid the dll boilerplate code for windows
 set(CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS ON)
-set(CMAKE_CXX_STANDARD 11)
+
+if (PARENT_CXX_STANDARD)
+    # Don't set CMAKE_CXX_STANDARD if it is already set by parent project
+    if (${PARENT_CXX_STANDARD} LESS 17)
+        message(FATAL_ERROR "cpr ${cpr_VERSION} does not support ${PARENT_CXX_STANDARD}. Please use cpr <= 1.9.0")
+    endif()
+else()
+    # Check and set supported c++ standard
+    if(CMAKE_CXX_COMPILER_ID STREQUAL "MSVC")
+        CHECK_CXX_COMPILER_FLAG("/std:c++17" COMPILER_SUPPORTS_CXX17)
+        CHECK_CXX_COMPILER_FLAG("/std:c++20" COMPILER_SUPPORTS_CXX20)
+    else()
+        CHECK_CXX_COMPILER_FLAG("-std=c++17" COMPILER_SUPPORTS_CXX17)
+        CHECK_CXX_COMPILER_FLAG("-std=c++20" COMPILER_SUPPORTS_CXX20)
+    endif()
+    if(COMPILER_SUPPORTS_CXX20)
+        set(CMAKE_CXX_STANDARD 20)
+    elseif(COMPILER_SUPPORTS_CXX17)
+        set(CMAKE_CXX_STANDARD 17)
+    else()
+        message(FATAL_ERROR "Compiler ${CMAKE_CXX_COMPILER} does not support C++17. Please use cpr <= 1.9.0")        
+    endif()
```

Comment:
Why all this logic. CPR shoult not set CXX standard to 20, if it is not set by the parent project.

Reply:
I thought the plan for cpr 1.10.0 was to use C++20 if supported and make C++17 required (like it is described here: https://github.com/libcpr/cpr/issues/382#issue-621811066)?

If I have misunderstood and this is not desired I can of course remove the additional logic.

Reply:
Yes, this was the plan since not all compilers are up to standard with cpp20.

Reply:
Ok, no problem :) I will look into this later and fix it.
Just one question: Should there still be a check with `CHECK_CXX_COMPILER_FLAG` if c++17 is supported. Otherwise I don't see an option how to raise an error message in the case that c++17 is not supported.

Reply:
There is no need for this, since CMake will should check this for us